### ISSUE TITLE
Docker Image CI Workflow failing

### ISSUE THREAD

**Issue by george-gca** [alshedivat/al-folio#2384]
### Have you checked that your issue isn't already filed?

- [X] I read through [FAQ](https://github.com/alshedivat/al-folio/blob/master/FAQ.md) and searched through the [past issues](https://github.com/alshedivat/al-folio/issues), none of which addressed my issue.
- [X] Yes, I have checked that this issue isn't already filed.

### Bug description

Docker Image CI Workflow is failing due to mini_racer

### How to reproduce the bug

Check the failed [action](https://github.com/alshedivat/al-folio/actions/runs/8928060333/job/24524664008).

### Error messages and logs

Error message of interest:

```
compiling mini_racer_extension.cc
linking shared-object mini_racer_extension.so
/usr/bin/ld: cannot find
/var/lib/gems/3.2.0/gems/libv8-node-21.7.2.0-x86_64-linux/vendor/v8/x86_64-linux-gnu/libv8/obj/libv8_monolith.a:
No such file or directory
collect2: error: ld returned 1 exit status
make: *** [Makefile:265: mini_racer_extension.so] Error 1

An error occurred while installing mini_racer (0.12.0), and Bundler cannot continue.
In Gemfile:
   mini_racer
```

### What operating system are you using?

Not applicable (e.g. you're using GitHub Pages or other hosting)

### Where are you seeing the problem on?

Deployed site

### More info

It is probably due to `mini_racer` version `0.12.0`. I checked their repo and found [this issue](https://github.com/rubyjs/mini_racer/issues/300). Can you take a look at it @pourmand1376?

**Comment by george-gca** [comment]
Update: just [asked on that bug](https://github.com/rubyjs/mini_racer/issues/300#issuecomment-2093711801) and they [pointed me to the cause](https://github.com/rubyjs/libv8-node/pull/36#issuecomment-2045274661).

**Comment by pourmand1376** [comment]
Hmm. Thanks for pointing out the issue. 

I think we should fix versions as suggested [here](https://stackoverflow.com/questions/9265213/should-i-specify-exact-versions-in-my-gemfile) to avoid such problem. 

This would require us to commit `Gemfile.lock` and we should install dependecies based on that. 

This way we would manually update our `Gemfile.lock` and if and only if, there is no problem with upstream branches, we update. 

And to fix this one, I think we may need to downgrade our Gemfile to not use the latest version of this library.

**Comment by george-gca** [comment]
I agree, we should commit `Gemfile.lock`. I believe it is safe to use the latest version of all dependencies, except `mini_racer`.

**Comment by wali-reheman** [comment]
> I agree, we should commit `Gemfile.lock`. I believe it is safe to use the latest version of all dependencies, except `mini_racer`.

Specify an older version of `mini_racer` doesn't work for me

**Comment by george-gca** [comment]
Can you test this combination?

```
Ruby 3.3.0
mini_racer 0.12.0
libv8-node 21.7.2.0
```

**Comment by samyakmehta28** [comment]
I am getting the same issue when I am running docker compose up command, what should I do to rectify it?

**Comment by pourmand1376** [comment]
Hi @george-gca,
I used `Gemfile.lock` and I tested both `0.12.0` versions and `0.9.0` versions and both of them give error. 

Also, we may have to provide two `Gemfile.lock` for solving the problem. 

Other than that, current users shouldn't have any problems. For your case, you shouldn't use `docker compose up`. But rather first pull the image via `docker compose pull` and then `docker compose up`.

**Comment by george-gca** [comment]
Why 2 `Gemfile.lock`? Also, how did you solve this if both are giving errors? Have you tried this combination? I found it [here](https://github.com/rubyjs/mini_racer/issues/300#issuecomment-2094066401).

```
Ruby 3.3.0
mini_racer 0.12.0
libv8-node 21.7.2.0
```

**Comment by betsybersson** [comment]
Is this conversation related to this build error: "Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: kitabisa/docker-slim-action@v1.0.3"

**Comment by george-gca** [comment]
It is not related, but I will open a PR for this. Thanks for noticing.

**Comment by broadcreate** [comment]
![image](https://github.com/alshedivat/al-folio/assets/34327137/21306917-8d8a-4195-a3c6-8cfe22e97850)

**Comment by george-gca** [comment]
Btw, using both my `Gemfile.lock` and the one currently in the repo throws this error when doing `docker compose build`:

```
23.97 current directory:
23.97 /var/lib/gems/3.2.0/gems/mini_racer-0.12.0/ext/mini_racer_extension
23.97 make DESTDIR\= sitearchdir\=./.gem.20240507-7-ifti4d
23.97 sitelibdir\=./.gem.20240507-7-ifti4d
23.97 compiling mini_racer_extension.cc
23.97 linking shared-object mini_racer_extension.so
23.97 /usr/bin/ld: cannot find
23.97 /var/lib/gems/3.2.0/gems/libv8-node-21.7.2.0-x86_64-linux/vendor/v8/x86_64-linux-gnu/libv8/obj/libv8_monolith.a:
23.97 No such file or directory
23.97 collect2: error: ld returned 1 exit status
23.97 make: *** [Makefile:265: mini_racer_extension.so] Error 1
23.97 
23.97 make failed, exit code 2
```

**Comment by amztc34283** [comment]
> Btw, using both my `Gemfile.lock` and the one currently in the repo throws this error when doing `docker compose build`:
> 
> ```
> 23.97 current directory:
> 23.97 /var/lib/gems/3.2.0/gems/mini_racer-0.12.0/ext/mini_racer_extension
> 23.97 make DESTDIR\= sitearchdir\=./.gem.20240507-7-ifti4d
> 23.97 sitelibdir\=./.gem.20240507-7-ifti4d
> 23.97 compiling mini_racer_extension.cc
> 23.97 linking shared-object mini_racer_extension.so
> 23.97 /usr/bin/ld: cannot find
> 23.97 /var/lib/gems/3.2.0/gems/libv8-node-21.7.2.0-x86_64-linux/vendor/v8/x86_64-linux-gnu/libv8/obj/libv8_monolith.a:
> 23.97 No such file or directory
> 23.97 collect2: error: ld returned 1 exit status
> 23.97 make: *** [Makefile:265: mini_racer_extension.so] Error 1
> 23.97 
> 23.97 make failed, exit code 2
> ```

Is this fixed with Docker?

**Comment by george-gca** [comment]
Maybe you need to update your repo. Probably some changes were made to the docker image that are affecting your local repo.

**Comment by amztc34283** [comment]
For anyone who does not want to rebase their branch for whatever reason, updating Dockerfile and Gemfile to v0.12.0 fixed this for me.

**Comment by tsnowak** [comment]
For those encountering this, the [change in this commit](https://github.com/alshedivat/al-folio/pull/2400/commits/850570e76a1d54eed0cb9f95f99d4eb47006aa9e) solved the issue for me.

For those that don't want to follow the link, that is, in the repo's `Dockerfile` change `ubuntu:latest` to `ruby:latest` (must just inherently apply the versioning fix others are suggesting above).

I have also set `platform: linux/amd64` in the `docker-compose.yaml` (I'm building on arm64 and wanted to force amd64), and I also added `RUN bundle lock --add-platform x86_64-linux-musl` to the `Dockerfile`. These changes might not be necessary, but in combination work.
